models/cartDao.js

```diff
@@ -0,0 +1,58 @@
+const appDataSource = require("./appDataSource");
+
+const getCartList = async (userId) => {
+  const [result] = await appDataSource.query(
+    `SELECT 
+    u.id as userId,
+    u.address,
+    JSON_ARRAYAGG(
+      JSON_OBJECT(
+        'cartId', c.id,
+        'productName', p.name,
+        'price', p.price,
+        'thumbnailImage', p.thumbnail_image,
+        'quantity', c.quantity)) AS cartList
+        from carts c
+        right join users u on u.id = c.user_id
+        left join products p on p.id = c.product_id
+        WHERE u.id = ?`,
+    [userId]
+  );
+  return result;
+};
+
+const addCartItems = async (userId, productId, quantity) => {
+  await appDataSource.query(
+    `INSERT INTO
+   carts
+   (user_id,
+    product_id,
+    quantity)
+   VALUES (?, ?, ?)
+   ON DUPLICATE KEY UPDATE 
+   user_id = ?, product_id =?, quantity = ?`,
+    [userId, productId, quantity, userId, productId, quantity]
+  );
+};
+
+const updateItemQuantity = async (userId, productId, quantity) => {
+  await appDataSource.query(
```

Comment:
```suggestion
const updateCart = async (userId, cartId,productId, quantity) => {
  return appDataSource.query(
```
cart를 업데이트 한다면 cartId도 프론트엔드에서 받아야합니다. 지금 userId와 productId를 기반으로 찾고있는데 보다 안전하고 확실한 방법은 cartId를 받는 것입니다.

또한 하나의 카트에 들어가는 상품은 한종류만 담기는 걸까요 아니면 여러종류가 함께 하나의 카트에 담길 수 있는걸까요? 전자의 경우라고 한다면 productId는 받을 필요 없이 cartId, quantity 만으로 가능 할 것 같습니다.

Reply:
네 알겠습니다 !